### DISCUSSION TITLE
No option handler

### DISCUSSION THREAD

**Discussion by ElDuderinoBerlin** [spectreconsole/spectre.console#1311] [Q&A] [answered]
Hi,

is there a "no-option-handler" to be used when no option is given so somebody can invoke an default option or handler/delegate?

Regards

**Comment by patriksvensson** [comment, marked as answer]
@maddinthegreat You can use the "Validate" overload of either Command<T> or CommandSettings to do this.

**Reply by rcdailey** [reply]
It isn't explicitly mentioned in the documentation that we should *mutate* the settings object in the Validate overload. I got the impression that would be an abuse of that feature as you aren't really validating anything, but setting a default value usually based on some runtime information (e.g. an injected service).
